sound/soc/sof/pm.c

```diff
@@ -218,6 +218,14 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	if (runtime_suspend)
 		sof_cache_debugfs(sdev);
 #endif
+
+	/* disable secondary cores */
+	ret = snd_sof_dsp_secondary_core_enable(sdev, BIT(0));
+	if (ret < 0) {
+		dev_err(sdev->dev, "failed to disable secondary cores\n");
+		return ret;
+	}
+
```

Comment:
I think this is wrong to expose this in in top-levle interface. Basicly here we have one platform where the host_managed_mask needs to be ignored at suspend and host needs to just go a power down the secondary cores. Right?

So I think this should a call in DSP-specific implementation, e.g. in hda_dsp_suspend() (or new variant just for tgl). Or if that doesn't work, add a new hook to ops to run some DSP code before power-down that TGL can then hook into.